[2025-03-10T18:45:16.821+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: project2.insert_employees manual__2025-03-10T18:45:12.658267+00:00 [queued]>
[2025-03-10T18:45:16.836+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: project2.insert_employees manual__2025-03-10T18:45:12.658267+00:00 [queued]>
[2025-03-10T18:45:16.837+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2025-03-10T18:45:16.838+0000] {taskinstance.py:1280} INFO - Starting attempt 1 of 1
[2025-03-10T18:45:16.838+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2025-03-10T18:45:16.856+0000] {taskinstance.py:1300} INFO - Executing <Task(PostgresOperator): insert_employees> on 2025-03-10 18:45:12.658267+00:00
[2025-03-10T18:45:16.863+0000] {standard_task_runner.py:55} INFO - Started process 110 to run task
[2025-03-10T18:45:16.870+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'project2', 'insert_employees', 'manual__2025-03-10T18:45:12.658267+00:00', '--job-id', '135', '--raw', '--subdir', 'DAGS_FOLDER/project2.py', '--cfg-path', '/tmp/tmpk7vyfv41']
[2025-03-10T18:45:16.871+0000] {standard_task_runner.py:83} INFO - Job 135: Subtask insert_employees
[2025-03-10T18:45:16.973+0000] {task_command.py:388} INFO - Running <TaskInstance: project2.insert_employees manual__2025-03-10T18:45:12.658267+00:00 [running]> on host f1d7771c7032
[2025-03-10T18:45:17.055+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=project2
AIRFLOW_CTX_TASK_ID=insert_employees
AIRFLOW_CTX_EXECUTION_DATE=2025-03-10T18:45:12.658267+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-03-10T18:45:12.658267+00:00
[2025-03-10T18:45:17.056+0000] {sql.py:254} INFO - Executing: CALL update_employees();
[2025-03-10T18:45:17.066+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2025-03-10T18:45:17.297+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2025-03-10T18:45:17.320+0000] {sql.py:375} INFO - Running statement: CALL update_employees();, parameters: None
[2025-03-10T18:45:17.339+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/operators/sql.py", line 266, in execute
    **extra_kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 349, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 380, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.NotNullViolation: null value in column "employee_id" of relation "employees" violates not-null constraint
DETAIL:  Failing row contains (null, 41, Female, Yes, Life Sciences, 3).
CONTEXT:  SQL statement "INSERT INTO employees (employee_id, age, gender, OverTime, EducationField, PerformanceRating)
SELECT s.employee_id, s.age, s.gender, s.OverTime, s.EducationField, s.PerformanceRating
FROM staging_hr s
LEFT JOIN employees e ON s.employee_id = e.employee_id WHERE e.employee_id IS NULL"
PL/pgSQL function update_employees() line 3 at SQL statement

[2025-03-10T18:45:17.351+0000] {taskinstance.py:1323} INFO - Marking task as FAILED. dag_id=project2, task_id=insert_employees, execution_date=20250310T184512, start_date=20250310T184516, end_date=20250310T184517
[2025-03-10T18:45:17.367+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 135 for task insert_employees (null value in column "employee_id" of relation "employees" violates not-null constraint
DETAIL:  Failing row contains (null, 41, Female, Yes, Life Sciences, 3).
CONTEXT:  SQL statement "INSERT INTO employees (employee_id, age, gender, OverTime, EducationField, PerformanceRating)
SELECT s.employee_id, s.age, s.gender, s.OverTime, s.EducationField, s.PerformanceRating
FROM staging_hr s
LEFT JOIN employees e ON s.employee_id = e.employee_id WHERE e.employee_id IS NULL"
PL/pgSQL function update_employees() line 3 at SQL statement
; 110)
[2025-03-10T18:45:17.402+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2025-03-10T18:45:17.421+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
